File: scala/bob/src/main/scala/Bob.scala
object Bob {
  def response(statement: String): String = {
    val input = statement.trim()
    val upper = statement.toUpperCase()
    val scream = upper == statement && statement.toLowerCase() != upper
    statement match {
      case _ if input == ""                 => "Fine. Be that way!"
      case _ if scream && input.last == '?' => "Calm down, I know what I'm doing!"
      case _ if input.last == '?'           => "Sure."
      case _ if scream                      => "Whoa, chill out!"
      case _                                => "Whatever."
    }
  }
}

File: fsharp/bob/Bob.fs
module Bob

let response (input: string): string =
    let inp = input.Trim()
    let question = inp.EndsWith("?")
    let upper = input.ToUpper()

    let scream =
        upper = input && input.ToLower() <> input

    match input with
    | _ when inp = ""           -> "Fine. Be that way!"
    | _ when scream && question -> "Calm down, I know what I'm doing!"
    | _ when question           -> "Sure."
    | _ when scream             -> "Whoa, chill out!"
    | _                         -> "Whatever."
